### PR TITLE
[node] add base64url BufferEncoding option

### PR DESCRIPTION
**Pull request by ermik**
New encoding option was introduced in nodejs/node#36952 and is available since v15.7.0. It is documented on the website: 
https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. **(does not)**

### CODE REVIEW

**Comment by ermik** [comment]
Happy to add tests if requested. Have not deemed them needed, but also did not see any existing tests for this feature. Also please let me know if CLA is needed for this. Thanks everybody.

**Comment by ermik** [comment]
`bent`, `cheerio`, and `mongoose` had test failures. After a cursory review I see no impact this PR could have on those failures. Let me know what I can do to assist.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Missing test(s).

**Comment by ermik** [comment]
Wow.

**Comment by peterblazejewicz** [comment]
> Happy to add tests if requested. Have not deemed them needed, but also did not see any existing tests for this feature. Also please let me know if CLA is needed for this. Thanks everybody.

nothing like that, but you're changing Node types, would be more than a nice addition,and this is promoted via self-merge option after maintainer approval, so win-win scenario)

**Pull request closed without merging**
